Tests/L2Tests/L2TestsPlugin/tests/Telemetry_L2Test.cpp

```diff
@@ -0,0 +1,484 @@
+{
+    JSONRPC::LinkType<Core::JSON::IElement> jsonrpc(TELEMETRY_CALLSIGN, TELEMETRYL2TEST_CALLSIGN);
+    StrictMock<AsyncHandlerMock_Telemetry> async_handler;
+    uint32_t status = Core::ERROR_GENERAL;
+    JsonObject params,thresholds;
+    struct _rbusObject rbObject;
+    struct _rbusValue rbValue;
+
+    JsonObject result;
+    uint32_t signalled = TELEMETRYL2TEST_STATE_INVALID;
+    std::string message;
+    JsonObject expected_status;
+
+    ON_CALL(*p_rBusApiImplMock, rbus_open(::testing::_, ::testing::_))
+        .WillByDefault(
+            ::testing::Return(RBUS_ERROR_SUCCESS));
+
+    ON_CALL(*p_rBusApiImplMock, rbusValue_GetString(::testing::_, ::testing::_))
+        .WillByDefault(
+            ::testing::Return( "SUCCESS"));
+
+    ON_CALL(*p_rBusApiImplMock, rbusObject_GetValue(::testing::_, ::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [&](rbusObject_t object, char const* name) {
+                EXPECT_EQ(object, &rbObject);
+                EXPECT_EQ(string(name), _T("UPLOAD_STATUS"));
+                return &rbValue;
+            }));
+
+    ON_CALL(*p_rBusApiImplMock, rbusMethod_InvokeAsync(::testing::_, ::testing::_, ::testing::_, ::testing::_, ::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [&](rbusHandle_t handle, char const* methodName, rbusObject_t inParams, rbusMethodAsyncRespHandler_t callback,  int timeout) {
+                callback(handle, methodName, RBUS_ERROR_SUCCESS, &rbObject);
+
+                return RBUS_ERROR_SUCCESS;
+            }));
+
+    /* errorCode and errorDescription should not be set */
+    EXPECT_FALSE(result.HasLabel("errorCode"));
+    EXPECT_FALSE(result.HasLabel("errorDescription"));
+
+    /* Register for temperature threshold change event. */
+    status = jsonrpc.Subscribe<JsonObject>(JSON_TIMEOUT,
+                                           _T("onReportUpload"),
+                                           &AsyncHandlerMock_Telemetry::onReportUpload,
+                                           &async_handler);
+
+    EXPECT_EQ(Core::ERROR_NONE, status);
+
+    message = "{\"telemetryUploadStatus\":\"UPLOAD_SUCCESS\"}";
+    expected_status.FromString(message);
+    EXPECT_CALL(async_handler, onReportUpload(MatchRequestStatus(expected_status)))
+        .WillOnce(Invoke(this, &Telemetry_L2test::onReportUpload));
+
+    //params["telemetryUploadStatus"] = "UPLOAD_SUCCESS";
+    status = InvokeServiceMethod("org.rdk.Telemetry.1", "uploadReport", params, result);
+    EXPECT_EQ(Core::ERROR_NONE, status);
+    EXPECT_FALSE(result["success"].Boolean());
+    EXPECT_STREQ("null", result["value"].String().c_str());
+
+
+    ON_CALL(*p_rBusApiImplMock, rbus_open(::testing::_, ::testing::_))
+        .WillByDefault(
+            ::testing::Return(RBUS_ERROR_SUCCESS));
+
+    EXPECT_CALL(*p_rBusApiImplMock, rbusMethod_InvokeAsync(::testing::_, ::testing::_, ::testing::_, ::testing::_, ::testing::_))
```

Comment:
In this expect call, test case is expecting first "RBUS_ERROR_SUCCESS" and then "RBUS_ERROR_BUS_ERROR", add a comment what is the reason.